### DISCUSSION TITLE
Get GTFS-RT of Transit Agency

### DISCUSSION THREAD

**Discussion by babangidazachariah** [transitland/transitland#405] [Q&A] [answered]
I am new to the world of GTFS and GTFS-RT. I found transitland and want to use it programmatically (python) retrieve GTFS-RT for a given transit agency. However, I am not sure how to get it done. Support showing code snippet would be appreciated. I am particularly interested in the realtime vehicle position and trip updates.

**Comment by drewda** [comment, marked as answer]
Transitland currently does not allow outside used to download full GTFS Realtime messages in protocol buffer. We have been thinking about adding this functionality and can share an update here if/when it's available. In the meantime, if you want the full GTFS Realtime feed contents, you will need to request it directly from an upstream feed source.

Currently, the Transitland v2 REST API provides information derived from GTFS Realtime under the stop departures endpoint: https://www.transit.land/documentation/rest-api/departures

Alerts derived from GTFS Realtime are also available on the agencies, operators, routes, trips, and stops endpoints. See the `include_alerts` query parameter on all those endpoints in the REST API: https://www.transit.land/documentation/rest-api
